Automattic/woocommerce-payments
# feat: tokenized ECE product page compatibility

Fixes part of https://github.com/Automattic/woocommerce-payments/issues/9722

#### Changes proposed in this Pull Request









Implementing the Tokenized ECE implementation on the product page.
The compatibility w/ WooCommerce Deposits is currently blocked due to https://github.com/woocommerce/woocommerce-deposits/issues/665

The changes have been ported from the Tokenized PRB implementation (in `client/tokenized-product-request`).
The tests are already present in the `ExpressCheckoutCartApi` and `ExpressCheckoutOrderApi`.

#### Testing instructions





- Update your WooPayments Dev Tools
- Enable the "Enable Cart-Token implementation for ECE" feature flag
- Restart your webpack build (`npm run start`)
- As a merchant, ensure you have GooglePay/ApplePay enabled in the WooPayments settings (for product pages, cart, checkout)
- As a customer, go to a product page
- Click on the GooglePay (or ApplePay, depending on your setup) dialog
- You should be able to interact with shipping addresses & shipping options (in case of a physical product)
- The "Order detail" dialog (the ℹ️ info icon next to the price) should only show the product on the page (and not other products that might be present on the cart)
- Selecting a different product quantity on the product page should be reflected on the dialog, once it is opened
- Selecting a different product variation (in case of variable products) should be reflected on the dialog, once it is opened
- You should be able to place an order for the selected product
- Once the order is placed, the main cart should remain unaffected
- The order page in `wp-admin` should reflect the cart contents, the currency being used, and the totals

-------------------


**Post merge**
